tla/consensus/ccfraft.tla

```diff
@@ -456,28 +456,49 @@ PlausibleSucessorNodes(i) ==
         highestMatchServers == {n \in activeServers : \A m \in activeServers : matchIndex[i][n] >= matchIndex[i][m]}
     IN {n \in highestMatchServers : \A m \in highestMatchServers: HighestConfigurationWithNode(i, n) >= HighestConfigurationWithNode(i, m)}
 
+\* Generate initial state for a given startNode
+StartState(startNode, servers) ==
+    /\ reconfigurationCount = 0
+    /\ removedFromConfiguration = {}
+    /\ configurations = [ i \in servers |-> IF i = startNode THEN (1 :> {startNode}) ELSE << >>]
+    /\ currentTerm = [i \in servers |-> IF i = startNode THEN 2 ELSE 0]
+    /\ state       = [i \in servers |-> IF i = startNode THEN Leader ELSE None]
+    /\ votedFor    = [i \in servers |-> Nil]
+    /\ log         = [i \in servers |-> IF i = startNode
+                                        THEN << [term |-> 2, contentType |-> TypeReconfiguration, configuration |-> {startNode}],
+                                                [term |-> 2, contentType |-> TypeSignature] >>
+                                        ELSE << >>]
+    /\ commitIndex  = [i \in servers |-> IF i = startNode THEN 2 ELSE 0]
+    /\ committableIndices  = [i \in servers |-> {}]
+
+\* Generate initial state for a startNodes configuration
+JoinedState(startNodes, servers) ==
+    /\ reconfigurationCount = 0
+    /\ removedFromConfiguration = {}
+    /\ \E startNode \in startNodes:
+        /\ configurations = [ i \in servers |-> IF i \in startNodes  THEN (3 :> startNodes) ELSE << >>]
+        /\ currentTerm = [i \in servers |-> IF i \in startNodes THEN 2 ELSE 0]
+        /\ state       = [i \in servers |-> IF i = startNode THEN Leader ELSE IF i \in startNodes THEN Follower ELSE None]
+        /\ votedFor    = [i \in servers |-> Nil]
+        /\ log         = [i \in servers |-> IF i \in startNodes
+                                            THEN << [term |-> 2, contentType |-> TypeReconfiguration, configuration |-> {startNode}],
+                                                    [term |-> 2, contentType |-> TypeSignature],
+                                                    [term |-> 2, contentType |-> TypeReconfiguration, configuration |-> startNodes],
```

Comment:
Why is it `{startNodes}` above and `startNodes` at log entry 3?

Reply:
Never mind, confused `startNode` and `startNodes`. => Former renamed to `sn` below.